src/index.c

```diff
@@ -148,6 +125,36 @@ static int write_index(git_oid *checksum, git_index *index, git_filebuf *file);
 static void index_entry_free(git_index_entry *entry);
 static void index_entry_reuc_free(git_index_reuc_entry *reuc);
 
+GIT_INLINE(int) index_map_set(git_index *idx, git_idxmap *map, git_index_entry *e)
+{
+	if (!map)
+		map = idx->entries_map;
```

Comment:
I found it a little odd that it took an `idx` and a `map` and did this switch.  Maybe splitting this into two functions?

`int index_map_set(git_idxmap *map, git_index_entry *e, bool ignore_case)`

and

`int index_entry_set(git_index *idx, git_index_entry *e)` which calls into that?  Names provisional, of course.

I have the same feelings about `index_map_resize`...

Reply:
Yeah, I know, I was pondering the same. I guess I'll just go with `index_map_set`, not even introducing a second `index_entry_set`. The additional boilerplate isn't even worth it to skip a single parameter, if you ask me. Will update in a few minutes